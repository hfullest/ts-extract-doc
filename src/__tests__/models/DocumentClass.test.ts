import { resolve } from 'path';
import { extractTsToMarkdown } from '../..';

describe('DocumentClass', () => {
  it('快照', () => {
    const result = extractTsToMarkdown(resolve(__dirname, '../fixtures/class.ts'));
    expect(result).toMatchSnapshot();
  });
});
