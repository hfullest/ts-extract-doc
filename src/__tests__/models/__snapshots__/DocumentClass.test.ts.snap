// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentClass 快照 1`] = `
"### ExampleClass

这个是对ExampleClass类的描述

#### 属性

|名称|说明|类型|默认值|版本|
|:--:|:--:|:--:|:---:|:--:|
|a |a属性|string|'124'|-|
|button ~(可选)~ |\`button\`描述  button的==内容==区|UiButtonProps|-|3.2.15|
|dddPublic |测试公共方法|number|12|-|

#### 方法

|名称|说明|类型|默认值|版本|
|:--:|:--:|:--:|:---:|:--:|
|bb |接口函数定义|(p1: string ,      p2?: number,   ) => string|() => ''|-|

#### 静态属性

|名称|说明|类型|默认值|版本|
|:--:|:--:|:--:|:---:|:--:|
|fffStaticProp |-|boolean|true|-|

#### 静态方法

|名称|说明|类型|默认值|版本|
|:--:|:--:|:--:|:---:|:--:|
|eeeStaticMethod |这个是静态方法|unknown|(a: number, b?: string) => {     return 456;   }|-|

"
`;
